### PR TITLE
libobs: add qt5 check for linux

### PR DESCRIPTION
**Pull request by kkartaltepe**
### Description
This adds a very poorman's check for qt5 linkages into obs. It works for
checking some of the common broken plugins on my machine but it may be
produce false positives if someone has the text `libQt5` in their
binary. It also slows down startup (by ~500ms) by introducing a bunch of shell/grep
invocations, but it prevents crashes on v28.

### Motivation and Context
Lots of linux users use plugins, as many as windows or more. But we didn't introduce this check so linux users just get a segfault and the experience sucks.

### How Has This Been Tested?

Loaded some of the commonly seen plugins that fail and observed they were not loaded. In the case where shell or grep is not available on the destination machine `system` should return an error and all errors are treated as `can_load=true`. Only when shell,grep, and the match succeed does `system` return `0` which we treat as `can_load=false`. So it should be fairly safe.

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by kkartaltepe** [comment]
there was some discussion over a version that just loads the binaries and checks what the loader pulled in. I shy away from that only because if someone/qt tries to be tricky things might fail. Also once we have qt5 symbols we dont know if they will conflict with anything and cause some problem later in runtime and my 5 minutes of testing simply didnt catch it. So i think the risk of a false positive is worth it (and it doesnt false positive on a pure obs setup so its at least deferred to the plugin authors who can ask us for help or read this PR).

**Comment by derrod** [comment]
If going the `system()` route, perhaps just checking `ldd` output for qt libs is a bit faster/saner?

**Comment by kkartaltepe** [comment]
> If going the `system()` route, perhaps just checking `ldd` output for qt libs is a bit faster/saner?

ldd is maybe a bit faster on my machine, but also is a very hairy shell script (will it fail somewhere unexpected?) that we have to parse the output of at the end or pipe through grep again. Id hope we remove this in the next version personally so leave it as simple as possible unless we find a false positive from this.

**Comment by kkartaltepe** [comment]
I went ahead and updated to use ldd as well. At least now there should be no false positives but maybe more false negatives.

**Comment by kkartaltepe** [comment]
From the discussion earlier, the specific timings for this are ~400-600ms added to startup for the 21 default plugins. For the alternate forking method discussed it adds ~200ms.

File: libobs/util/platform-nix.c [hunk 1]
```diff
@@ -99,7 +100,13 @@ void os_dlclose(void *module)
 void get_plugin_info(const char *path, bool *is_obs_plugin, bool *can_load)
 {
 	*is_obs_plugin = true;
+#if OBS_QT_VERSION == 6 && !defined(__APPLE__)
+	char cmd[2048];
+	snprintf(cmd, 2048, "ldd \"%s\" | grep -q 'libQt5'", path);
```
[the remark this answers is not in the document]
Comment (kkartaltepe) [reply libobs/util/platform-nix.c]: Good idea, done.

File: libobs/util/platform-nix.c [hunk 2]
```diff
@@ -99,7 +100,13 @@ void os_dlclose(void *module)
...
+	*can_load = !!system(cmd);
+#else
 	*can_load = true;
+#endif
 	UNUSED_PARAMETER(path);
```
[the remark this answers is not in the document]
Comment (kkartaltepe) [reply libobs/util/platform-nix.c]: Done.

**Comment by tytan652** [comment]
Can be closed because #7342 got merged.

**Comment by derrod** [comment]
Closing as per above.

**Pull request closed without merging**